### PR TITLE
hb-07 is done

### PR DESCRIPTION
**Pull request by vladkuzya**

### CODE REVIEW

File: src/main/java/mate/academy/dao/impl/OrderDaoImpl.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            return session.createQuery("SELECT DISTINCT o FROM Order o "
+                    + "JOIN FETCH o.tickets "
+                    + "WHERE o.user = :user", Order.class)
+                    .setParameter("user", user)
+                    .getResultList();
+        } catch (Exception e) {
+            throw new DataProcessingException("Error retrieving order history for user "
+                    + user.toString(), e);
+        }
```
Comment (dariiabutorina) [inline src/main/java/mate/academy/dao/impl/OrderDaoImpl.java]: Is there a need to catch an exception with the _**READ**_ operation?

File: src/main/java/mate/academy/service/impl/OrderServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    private OrderDao orderDao;
+
+    @Override
+    public Order completeOrder(List<Ticket> tickets, User user) {
+        Order order = new Order();
+        order.setTickets(tickets);
+        order.setUser(user);
+        order.setOrderData(LocalDateTime.now());
+        orderDao.add(order);
+        return order;
```
Comment (dariiabutorina) [inline src/main/java/mate/academy/service/impl/OrderServiceImpl.java]: 1. Can we inline it?
2. What should happen to the shopping cart's tickets after completing the order?

Comment (vladkuzya) [reply src/main/java/mate/academy/service/impl/OrderServiceImpl.java]: clear()

File: src/main/java/mate/academy/dao/impl/OrderDaoImpl.java (hunk 1 above)
Comment (vladkuzya) [reply src/main/java/mate/academy/dao/impl/OrderDaoImpl.java]: no of course

**Review by KseniiaMakarova — APPROVED** [review]

**Pull request merged**